common/src/View/MapDocument.cpp

```diff
@@ -1568,7 +1568,7 @@ namespace TrenchBroom {
         }
         
         void MapDocument::loadTextures() {
-            texturesWillBeUnloadedNotifier();
+            Notifier0::NotifyBeforeAndAfter notifyTextureCollections(textureCollectionsWillChangeNotifier, textureCollectionsDidChangeNotifier);
```

Comment:
These notifiers usually only get called from `MapDocumentCommandFacade`. Sorry, should have mentioned this earlier, but these methods may be called in situations where notifications are not necessary, so it's safer to move the notifications down to the `perform*` methods in *MapDocumentCommandFacade`.